egs2/TEMPLATE/asr1/scripts/feats/perform_kmeans.sh

```
@@ -260,14 +351,49 @@ if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ] && ! [[ " ${skip_stages} " =~ [
+                          scp = dir "/inference_kmeans." k ".scp"
+                          u2n = dir "/utt2num_samples." k }
+                {
+                  print           > scp
+                  print $1, n[$1] > u2n
+                  acc += n[$1]
+                  if (k < nj && acc >= k * target) {
+                      close(scp); close(u2n); k++
+                      scp = dir "/inference_kmeans." k ".scp"
+                      u2n = dir "/utt2num_samples." k
+                  }
+                }' ${datadir}/${dset}/utt2num_samples "${key_file}"
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)